[SC-316] Zapisywanie ankiety przez studenta

barto14753 opened the pull request:
Wywaliłem UserFeedback i wszystko co z nim związane bo SurveyResult jest wykorzystywany już w wielu miejscach.  Dostępne endpointy:
- POST /survey/result -> zapisujemy odpowiedź lub modyfikujemy (punkty za ankietą są przyznawane raz)
- GET /survey/result

 i jeszcze teraz w GET /survey jeśli wywoła go student to dostanie też dane o jego odpowiedzi na ankiete

Zmodyfikowałem już istniejący POST /feedback/user i dodałem GET'a. W obecnej wersji ponowne wysłanie POSTA edytuje feedback. I jeszcze dodałem do GET /survey pole z feedbackiem jeśli student już udzielił odpowiedzi.

ajablonsk1 committed:
[SC-297] Added badges and unlockedBadge entities. Added 19 exemplary badges.

ajablonsk1 committed:
[SC-297] Added badges and BadgeVisitor

ajablonsk1 committed:
[SC-297] Implementing badge visitor

ajablonsk1 committed:
[SC-297] Merge remote-tracking branch 'origin/develop' into SC-297-endpoint-na-odznaki-oraz-na-ich-sprawdzenie

# Conflicts:
#	api/src/main/java/com/example/api/config/DatabaseConfig.java
#	api/src/main/java/com/example/api/model/user/User.java

ajablonsk1 committed:
[SC-297] Now badges are checked every time student gets points from taskResult

ajablonsk1 committed:
[SC-297] Badges should work now

ajablonsk1 committed:
[SC-297] Little refactor and mistakes are fixed

ajablonsk1 committed:
[SC-297] Merge remote-tracking branch 'origin/develop' into SC-297-endpoint-na-odznaki-oraz-na-ich-sprawdzenie

ajablonsk1 committed:
[SC-297] Changed a little badge response

barto14753 committed:
[SC-316] Adjusted submitting survey

barto14753 committed:
[SC-316] Fixed assigning points for survey

barto14753 committed:
[SC-316] Added endpoint for getting user feedback by student

barto14753 committed:
[SC-316] Fixed rate validation and UserFeedbackRepo

barto14753 committed:
[SC-316] Replaced UserFeedback with SurveyResult

ajablonsk1 reviewed (approved):
Spawdzilem u siebie i wyglada jakby dzialalo

barto14753 committed:
[SC-319] Dodanie wyników ankiety do listy z punktami (#244)

barto14753 committed:
[SC-316] Merge branch 'develop' into SC-316-zapisywanie-ankiety-przez-studenta

barto14753 committed:
[SC-316] Merge branch 'develop' into SC-316-zapisywanie-ankiety-przez-studenta

# Conflicts:
#	api/src/test/java/com/example/api/unit/service/activity/feedback/UserFeedbackServiceTest.java

barto14753 committed:
[SC-316] Removed UserFeedbackServiceTest
